### DISCUSSION TITLE
Device Types!

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/devicetype-library#1347] [Q&A]
So I have a NetBox in my private environment and A lot of device types are missing from the library , is there a way to contribute ?

Also I have a question regarding N9K-C9504 Chassis -
There is a yaml file available of this device types but it has no interfaces other then mgmt0 and mgmt1 and If I want to add more interfaces I need to use QSFP-100G40G-BIDI which is not in the interface types and I'm lost! What should I do?

(I have uploaded the devicetype-library into my environment)

**Comment by danner26** [comment]
Absolutely. To contribute to this repo all you have to do is:

1. For this repo to your personal account
2. Create a new branch inside of your fork for the changes
3. Make your changes and push them to GitHub
4. Create a PR from your forks branch to the master branch on this upstream repo

In regards to the chassis, since it is a chassis device it should have modules slot into it. The associated module bays should be added if they are not in that file (to do so, follow the instructions above and just modify the already existing file) and then the associated modules should be created as well.

If you have anymore questions just let us know!
